File: cpp/Problem492.cpp
/*
 * Problem: 492. Construct the Rectangle
 * URL: https://leetcode.com/problems/construct-the-rectangle/
 * Difficulty: Easy
 * Tags: Math
 */

#include <cmath>
#include <vector>

using namespace std;

class Solution {
public:
	vector<int> constructRectangle(int area) {
		vector<int> results(2, 0);
		for (int i = (int)sqrt(area); i >= 1; i--) {
			if (area % i == 0) {
				results[0] = area / i;
				results[1] = i;
				break;
			}
		}
		return results;
	}
};

File: python/Problem492.py
# Problem: 492. Construct the Rectangle
# URL: https://leetcode.com/problems/construct-the-rectangle/
# Difficulty: Easy
# Tags: Math

import math
from typing import List

class Solution:
  def constructRectangle(self, area: int) -> List[int]:
    results: List[int] = [0] * 2
    for i in range(int(math.sqrt(area)), 0, -1):
      if area % i == 0:
        results[0] = area // i
        results[1] = i
        break
    return results